### PR TITLE
Fix module definitions for `pkg/util/{cgroups,log,scrubber}`

### PR DESCRIPTION
**Pull request by paulcacheux**
### What does this PR do?

This PR fixes the definition of the new modules following the documentation https://github.com/DataDog/datadog-agent/blob/main/docs/dev/modules.md. This will allow them to be included in functions like `inv -e tidy-all`, tests etc.

This PR also cleans up the `go.mod` file

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature. 
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: tasks/modules.py [hunk 1]
```diff
@@ -117,6 +118,9 @@ def dependency_path(self, agent_version):
     "pkg/otlp/model": GoModule("pkg/otlp/model", independent=True),
     "pkg/security/secl": GoModule("pkg/security/secl", independent=True),
     "pkg/remoteconfig/state": GoModule("pkg/remoteconfig/state", independent=True),
+    "pkg/util/cgroups": GoModule("pkg/util/cgroups", condition=lambda: sys.platform == "linux"),
+    "pkg/util/log": GoModule("pkg/util/log"),
```
Comment (KSerrania) [inline tasks/modules.py:122]: According to `pkg/util/log/go.sum` and `pkg/util/cgroups/go.sum`, `pkg/util/log` and `pkg/util/cgroups` seem independent from the main `datadog-agent` module, why couldn't we add `independent=True` to them here?

**Review by clamoriniere — APPROVED** [review]

**Review by KSerrania — APPROVED** [review]

File: docs/dev/modules.md [hunk 2]
```diff
@@ -25,16 +25,15 @@ After you have refactored, if needed, and listed the packages that you want to e
  	    github.com/DataDog/datadog-agent/path/to/module => ./path/to/module
     )
     ```
-1. Update the `DEFAULT_MODULES` dictionary on the `tasks/modules.py` file. You need to:
-    - create a new module, specifying the path, targets, dependencies and a condition to run tests (if any) and
-    - update the dependencies of other modules if they depend on this one.
+1. Update the `DEFAULT_MODULES` dictionary on the `tasks/modules.py` file. You need to create a new module, specifying the path, targets and a condition to run tests (if any).
```
Comment (maycmlee) [inline docs/dev/modules.md:28]:
```suggestion
1. Update the `DEFAULT_MODULES` dictionary in the `tasks/modules.py` file. You need to create a new module, specifying the path, targets, and a condition to run tests (if any).
```

**Review by maycmlee — COMMENTED** [review]
Left a couple of small edits!

**Review by maycmlee — APPROVED** [review]

**Review by djmitche — APPROVED** [review]
I'm no expert on go.mod but the docs change looks good to me.  I'll get this merged in the go1.18 PR when it lands.

**Pull request merged**